File: mobileBE/src/main/java/com/uth/mobileBE/dto/request/ReaderRequest.java
package com.uth.mobileBE.dto.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReaderRequest {
    private String fullName;
    private String phone;
    private String barcode;
    private Boolean isStudent;
    private Long libraryId;
    private LocalDateTime membershipExpiry;
}

File: app/src/main/java/com/example/quanlythuvien/data/model/request/ReaderRequest.kt
package com.example.quanlythuvien.data.model.request

data class ReaderRequest(
    val fullName: String,
    val phone: String,
    val barcode: String,
    val libraryId: Long,
    val monthRegis: Long = 0L,
    val membershipExpiry: String? = null,
    val isBlocked: Boolean? = null
)